### PR TITLE
qemu..smbios: q35 and py3 fixes

### PR DESCRIPTION
**Pull request by ldoktor**
This contains some config tweaks to allow running this test under q35 profile, then some py3 fixes and adjustment to avoid booting unwanted upstream targets.

@KarMez-Redhat, @wainersm, @yashkmankad: this should help you with your CI.

### CODE REVIEW

File: qemu/tests/cfg/smbios.cfg [hunk 1]
```diff
@@ -5,6 +5,10 @@
     type = smbios_table
     requires_root = yes
     start_vm = no
+    # Remove extra pci_controllers as they only make sense for q35 but we
+    # might be booting i440fx machines as well which would require PCIE
+    # bridge
+    pcie_extra_root_port = 0
```
Comment (vivianQizhu) [inline qemu/tests/cfg/smbios.cfg]: Yes, will get error on plugging pcie-root-port to pci.0 when boot with `-machine pc-i440fx-rhel7.6.0` in test case `smbios_table.machine_type.q35`.

**Comment by vivianQizhu** [comment]
L#42 need decode too, and L#89 does not need decode, session returns string instead of bytes.

**Comment by ldoktor** [comment]
Oh, true. I only addressed the place that was failing for me. Let me update...

**Comment by ldoktor** [comment]
And yes, that looks like a product bug, upstream qemu passes...

**Comment by ldoktor** [comment]
Actually even upstream fails with several different messages. Anyway these are qemu issues and not ours (hopefully).

**Comment by vivianQizhu** [comment]
@ybduan Request for your ack, thanks.

**Comment by ybduan** [comment]
Sorry for the late response as I am not familiar with this automation case.

In fact, the failure is not a product bug and it is introduced by Line 61 @ldoktor .
The elements in list "support_machine_types" and list "expect_system_versions" are in one-to-one relationship as column 1&2 of "/usr/libexec/qemu-kvm -M ?" output.
But after masking isapc/xenpv/xenfv at line 61, the element order in list "support_machine_types" is changed.
This relationship is used at line 100&101.

Thanks!

**Comment by ldoktor** [comment]
Oups, I'm sorry, you are right. I wanted to use "smarter" approach, but let's use simple "if sm_type in (...); continue".... I'll re-test and push new version once I'm sure it works well.

**Comment by ldoktor** [comment]
Well still getting failures, but now they look like qemu issues (or simply different expectations of downstream and upstream version):

```
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-3.0'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-2.9'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-2.8'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-2.7'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-2.6'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-2.5'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-2.4'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-2.3'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-2.2'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-2.12'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-2.11'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-2.10'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-2.1'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'pc-i440fx-2.0'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996) (deprecated)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (i440FX + PIIX, 1996) (deprecated)' but guest is : 'ThinkPad T480s'
System.Version mismatch, Set 'Standard PC (Q35 + ICH9, 2009)' but guest is : 'pc-q35-3.0'
System.Version mismatch, Set 'Standard PC (Q35 + ICH9, 2009)' but guest is : 'pc-q35-2.9'
System.Version mismatch, Set 'Standard PC (Q35 + ICH9, 2009)' but guest is : 'pc-q35-2.8'
System.Version mismatch, Set 'Standard PC (Q35 + ICH9, 2009)' but guest is : 'pc-q35-2.7'
System.Version mismatch, Set 'Standard PC (Q35 + ICH9, 2009)' but guest is : 'pc-q35-2.6'
System.Version mismatch, Set 'Standard PC (Q35 + ICH9, 2009)' but guest is : 'pc-q35-2.5'
System.Version mismatch, Set 'Standard PC (Q35 + ICH9, 2009)' but guest is : 'pc-q35-2.4'
System.Version mismatch, Set 'Standard PC (Q35 + ICH9, 2009)' but guest is : 'pc-q35-2.12'
System.Version mismatch, Set 'Standard PC (Q35 + ICH9, 2009)' but guest is : 'pc-q35-2.11'
System.Version mismatch, Set 'Standard PC (Q35 + ICH9, 2009)' but guest is : 'pc-q35-2.10'
```
But using downstream RHEL qemu it passes:

```
avocado.test: PASS 1-type_specific.io-github-autotest-qemu.smbios_table.machine_type
avocado.test: 
PASS (132.42 s)
```

**Comment by ybduan** [comment]
Do we need set "only Host_RHEL"? @ldoktor

**Comment by ldoktor** [comment]
I'd hate that. How about this version?

**Comment by ybduan** [comment]
LGTM.
Acked-by: Yanbin Duan yduan@redhat.com

**Comment by ldoktor** [comment]
@vivianQizhu would you please re-review this updated version? It should work well on RHEL as well with upstream version.

**Pull request merged**